src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/LocationRepository.cs

```diff
@@ -0,0 +1,315 @@
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        break;
+                    case LocationProviderOptionMeta providerOption:
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendValue(providerOption.Ukprn);
+                        break;
+                    case LocationSchoolOptionMeta schoolOption:
+                        insertRow.AppendNullValue();
+                        insertRow.AppendNullValue();
+                        insertRow.AppendValue(schoolOption.Urn);
+                        insertRow.AppendValue(schoolOption.LaEstab);
+                        insertRow.AppendNullValue();
+                        break;
+                }
+
+                insertRow.EndRow();
+            }
+        }
+    }
+
+    private static List<GeographicLevel> ListLocationLevels(IReadOnlySet<string> allowedColumns)
+    {
+        return
+        [
+            .. allowedColumns
+                .Where(CsvColumnsToGeographicLevel.ContainsKey)
+                .Select(col => CsvColumnsToGeographicLevel[col])
+                .Distinct()
+                .OrderBy(EnumToEnumLabelConverter<GeographicLevel>.ToProvider)
+        ];
+    }
+
+    private static LocationOptionMeta MapLocationOptionMeta(
+        IDictionary<string, string> row,
+        GeographicLevel level)
+    {
+        var cols = level.CsvColumns();
+        var label = row[level.CsvNameColumn()];
+
+        return level switch
+        {
+            GeographicLevel.LocalAuthority => new LocationLocalAuthorityOptionMeta
+            {
+                PublicId = string.Empty,
+                Label = label,
+                Code = row[LocalAuthorityCsvColumns.NewCode],
+                OldCode = row[LocalAuthorityCsvColumns.OldCode]
+            },
+            GeographicLevel.School => new LocationSchoolOptionMeta
+            {
+                PublicId = string.Empty,
+                Label = label,
+                Urn = row[SchoolCsvColumns.Urn],
+                LaEstab = row[SchoolCsvColumns.LaEstab]
+            },
+            GeographicLevel.Provider => new LocationProviderOptionMeta
+            {
+                PublicId = string.Empty, Label = label, Ukprn = row[ProviderCsvColumns.Ukprn]
+            },
+            GeographicLevel.RscRegion => new LocationRscRegionOptionMeta
+            {
+                PublicId = string.Empty, Label = label
+            },
+            _ => new LocationCodedOptionMeta
+            {
+                PublicId = string.Empty, Label = label, Code = row[cols.Codes.First()]
+            }
```

Comment:
Would multi-line all these to make them more readable. 

Btw, do you have some option enabled that's folding these for you automatically? Feels like it might be doing it too aggressively!

Reply:
Sorted my formatting settings now. I had it setup to 'chop always' which I thought would chop every element, but I didn't realise that setting works in conjunction with the 'max object and collection intializer elements on a single line' which was set to 4. 😆